### ISSUE TITLE
JsonPrimitive property names incompatible with Objective-C

### ISSUE THREAD

**Issue by nasahapps** [Kotlin/kotlinx.serialization#306]
The [JsonPrimitive](https://github.com/Kotlin/kotlinx.serialization/blob/0dafc0f2055a2e76885101a951ad331334400a31/runtime/common/src/main/kotlin/kotlinx/serialization/json/JsonElement.kt) class contains many properties whose names are the exact same as their type (int, double, float, etc.), like so:

```kotlin
val int: Int 
    get() = content.toInt()
val long: Long 
    get() = content.toLong()
val double: Double 
    get() = content.toDouble()
val float: Float 
    get() = content.toFloat()
```

 Although this works in Kotlin due to Kotlin types being capitalized, when this code is compiled for native Objective-C usage, because those types are lowercase, they're considered type names and not property names. This results in something like this in the generated header file:

```objc
__attribute__((swift_name("Kotlinx_serialization_runtime_nativeJsonPrimitive")))
@interface SharedKotlinx_serialization_runtime_nativeJsonPrimitive : SharedKotlinx_serialization_runtime_nativeJsonElement
@property (readonly) double double;
@property (readonly) float float;
@property (readonly) int32_t int;
@property (readonly) int64_t long;
@end;
```

and the program can't compile because of this.

KotlinX Serialization version: 0.9.1
Kotlin version: 1.3.11
Gradle version: 4.7

**Comment by elizarov** [comment]
This looks like a Kotlin/Native issue (it should screen/mangle such names)

**Comment by SvyatoslavScherbina** [comment]
> it should screen/mangle such names

I don't think so: Kotlin/JVM doesn't apply any mangling in similar cases.

**Comment by nasahapps** [comment]
@sandwwraith The workaround from https://github.com/JetBrains/kotlin-native/issues/1898#issuecomment-420260369 worked, thanks.

Also, I noticed that issue should've been fixed in [this commit](https://github.com/JetBrains/kotlin-native/commit/132e12ba3e0ff12d975cdaee34b72bee88928153) which is tagged as part of Kotlin Native 1.1. I tried using `org.jetbrains.kotlin:kotlin-native-gradle-plugin:1.3.20-eap-15-native-1.1.0-rc1-5227` thinking it had KN 1.1, but either it doesn't or my setup wasn't configured properly.
